chanzuckerberg/corpora-data-portal
# frontend dataset view

#675 

https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/675

Adds dataset view functionality to the collection view
Shows upload status and dataset metadata when available

#### Reviewers
**Functional:** 
@tihuan 
**Readability:** 
@MDunitz 

NOTES:
1. Please note that the checkbox will be replaced with radio button when we implement download files in [Part 3](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/705undefined)
2. The Download button is not functional until [Part 3](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/705undefined) is complete too

---

## Changes
* `frontend/src/common/API.ts`
  * Add status endpoint
 * `frontend/src/common/entities.ts`
   * Add entities for dataset upload status 
 * `frontend/src/common/queries/datasets.ts` + `frontend/src/common/queries/entities.ts`
    * Add query for dataset status
  * `frontend/src/components/Collections/components/Grid/components/DatasetsGrid/index.tsx`
    * create grid for viewing datasets 
  * `frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/index.tsx`
    * Dataset row component, fetches metadata and upload status
  * `frontend/src/components/DropboxChooser/index.tsx` + `frontend/src/views/Collection/index.tsx`
    * Include filename in dropbox chooser
   * `frontend/src/views/Collection/components/DatasetUploadToast/index.tsx`
     * Create toast for communicating dataset upload
    
---

### QA Steps

1. Navigate to a private collection and Follow CTA to upload dataset
   1. Dataset should appear on grid and show uploading status
   1. It will take several minutes for any progress to be shown 
1. At successful upload a toast will appear
   1. Validation process indicator would be shown now, but backend is not currently communicating this
1. After validation and conversion completes, explore button and metadata should populate

1. Erroneous datasets will error and communicate error state via icon in title column
   1. Validation errors are currently communicated as upload error due to backend 


---
![image](https://user-images.githubusercontent.com/8716829/104073397-6e3e1600-51c2-11eb-9049-404bdb97f52d.png)

![image](https://user-images.githubusercontent.com/8716829/104073382-5f576380-51c2-11eb-9770-e0f674ef3845.png)

![image](https://user-images.githubusercontent.com/8716829/104073339-4484ef00-51c2-11eb-8042-1190294e478e.png)
